nodeup/pkg/model/network.go

```diff
@@ -1,123 +0,0 @@
-/*
-Copyright 2019 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package model
-
-import (
-	"fmt"
-	"path/filepath"
-
-	"k8s.io/klog"
-	"k8s.io/kops/pkg/systemd"
-	"k8s.io/kops/upup/pkg/fi"
-	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
-)
-
-// NetworkBuilder writes CNI assets
-type NetworkBuilder struct {
-	*NodeupModelContext
-}
-
-var _ fi.ModelBuilder = &NetworkBuilder{}
-
-// Build is responsible for configuring the network cni
-func (b *NetworkBuilder) Build(c *fi.ModelBuilderContext) error {
-	var assetNames []string
-
-	// @TODO need to clean up this code, it isn't the easiest to read
-	networking := b.Cluster.Spec.Networking
-	if networking == nil || networking.Classic != nil {
-	} else if networking.Kubenet != nil || networking.GCE != nil {
-		assetNames = append(assetNames, "bridge", "host-local", "loopback")
-	} else if networking.External != nil {
```

Comment:
Where did the asset adding for External go? Is KubenetBuilder supposed to also act for networking.External?

Reply:
External is a bit special. Kops doesn't do anything here, but expects a daemonset to handle networking. But it is a good point that we should probably just install the plugins required by kubenet anyway.

Reply:
This is replaced with a generic check for kubenet-based providers now.

Reply:
The comment immediately below says "external is based on kubenet". `pkg/model/components.UsesKubenet()` returns `true` for External.

The lack of documentation makes it unclear to me what External is for. I'm not confident one could, for example, run a Cilium daemonset with networking set to External.

Reply:
Comment resolved by subsequent commit (as `UsesKubenet()` returns `true` for External).

Reply:
So external baffled me a bit as well.
Kublet has a flag `network-plugin` that can for all intents and purposes only be set to `kubenet` or `cni`. Cilium, calico etc are CNI plugins. Kubenet is an older implementation where kubernetes handled more stuff internally, but you could still provide an additional daemonset to create overlay networks and such (e.g kopeio vxlan).

on kops `--networking cni` means bring your own cni plugin, while `--networking external` means bring your own kubenet "plugin". So it basically just determines what value is passed to kublet.

I plan on cleaning up our docs wrt this in a later PR.

Reply:
For this future doc cleanup:

When I removed Classic from `docs/networking.md`, that reduced the first section to stating that there are 3 networking modes. I think that needs to be taken down to 2 modes: "Kubenet" and "CNI". External and CNI should be put into their own consolidated "BYO" section.

That doc doesn't adequately mention the Kubenet-based variants and I don't have the knowledge to add them. It's also not clear to me whether the Kubenet info in that doc applies to all Kubenet variants or just to the Kubenet provider itself.